### PR TITLE
Add Classes and Imports to the AST

### PR DESCRIPTION
**Pull request by James300**
Associated Issue: #4151

This is the first step towards getting React icons displaying for their appropriate tab in the debugger.

### Summary of Changes

* AST grabs imports and classes
* Framework check for React Component
* Added test for React Component

### CODE REVIEW

**Comment by jasonLaster** [comment]
Nice @James300!

File: src/utils/parser/frameworks.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        }
+      });
+    }
+  });
+
+  return false;
+}
+
+function extendsComponent(classes) {
+  classes.forEach(classObj => {
```
Comment (jasonLaster) [inline src/utils/parser/frameworks.js]: lets try `Array.prototype.any` or `some` or lodash any/some

File: src/utils/parser/frameworks.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  if (!importsReact(imports) || !extendsComponent(classes)) {
+    return false;
+  }
+
+  return true;
+}
+
+export function isReduxAction(source) {
+  const { functions } = getSymbols(source);
```
Comment (jasonLaster) [inline src/utils/parser/frameworks.js]: lets drop these functions and just do the `isReactComponent` work now

File: src/utils/tests/framework.spec.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import * from "../frameworks";
+import { getSource } from "./helpers";
+
+describe("Parser.frameworks", () => {
+  it("isReactComponent", () => {
+    const symbols = ;
+      expect(isReactComponent(getSource("frameworks/component"))).toBe(true)();
```
Comment (jasonLaster) [inline src/utils/tests/framework.spec.js]: this is awesome if it passes :)

File: src/utils/tests/framework.spec.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  });
+});
+
+describe("framework", () => {
```
Comment (jasonLaster) [inline src/utils/tests/framework.spec.js]: lets drop this for now as we don't need redux stuff

File: src/utils/parser/frameworks.js (hunk 2 above)
Comment (James300) [reply src/utils/parser/frameworks.js]: Will do 👍

File: src/utils/parser/tests/__snapshots__/getSymbols.spec.js.snap [hunk 5]
```diff
@@ -394,73 +411,81 @@ Array [
...
+[(38, 20), (38, 28)]   sayHello  ",
   "memberExpressions:
- [(23, 18), (23, 29)] [(23, 10), (23, 29)] name.undefined.toUpperCase toUpperCase  
-[(23, 15), (23, 16)] [(23, 10), (23, 17)] name. undefined  
-[(23, 39), (23, 48)] [(23, 34), (23, 48)] name.substring substring  
-[(28, 4), (28, 8)] [(25, 19), (28, 8)]  join  
-[(27, 4), (27, 7)] [(25, 19), (27, 7)]  map  
-[(26, 4), (26, 7)] [(25, 19), (26, 7)] map map  
-[(30, 15), (30, 24)] [(30, 2), (30, 24)] globalObject.greetings greetings  
-[(38, 11), (38, 14)] [(38, 3), (38, 14)] console.log log  ",
```
Comment (jasonLaster) [inline src/utils/parser/tests/__snapshots__/getSymbols.spec.js.snap]: yeah... this _is_ pretty weird...

Comment (jasonLaster) [inline src/utils/parser/tests/__snapshots__/getSymbols.spec.js.snap]: could you rebase and run `yarn nom` this looks like some windows CLRF stuff

**Comment by James300** [comment]
Looks like one of the integration tests broke, I didn't think my work would have an effect on anything that covers yet so I didn't run the mochi test locally. I'll fix that tonight.

**Comment by James300** [comment]
@jasonLaster It looks like the tests that are failing are failing on master as well, can you confirm this please? If so, then I believe this PR is ready unless you can think of anything else to add.

**Pull request merged**